cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -79,3 +79,13 @@ set(PASM_FLAGS -x assembler-with-cpp -E -P)
 
 set(SPP_CMD ${CMAKE_C_COMPILER})
 set(SPP_FLAGS -x assembler-with-cpp -E -P)
+
+#Configure the platform dependent library for multithreading
+#TODO: we dont actually have a clang config and use gnu on non-Windows,
+#so we have to detect Apple Clang here.
+#see ../../OmrDetectSystemInformation.cmake
+if(CMAKE_C_COMPILER_ID MATCHES "^(Apple)?Clang$")
+	set(THREAD_LIBRARY pthread)
```

Comment:
Out of consistency, this variable should be called `OMR_PLATFORM_THREAD_LIBRARIES`, and we need to assert that this variable is defined in verify.cmake.

Also, it was a deliberate decision to classify any gnuish toolchain as "gnu", so this isn't really a TODO. If we want to revisit that decision, I'm open for discussing alternatives.